core/src/test/scala/com/pingcap/tispark/datasource/BaseDataSourceTest.scala

```diff
@@ -26,6 +26,10 @@ class BaseDataSourceTest(val table: String,
   override def beforeAll(): Unit = {
     enableTidbConfigPropertiesInjectedToSpark = _enableTidbConfigPropertiesInjectedToSpark
     super.beforeAllWithoutLoadData()
+    if (!database.equals("tispark_test")) {
+      tidbStmt.execute(s"drop database if exists $database")
+      tidbStmt.execute(s"create database $database")
+    }
```

Comment:
why recreate the database?

Reply:
what if database = tpch_test?

Reply:
It is not recreating database but creates a new database. In our BaseDataSourceSuite, the default database is `tispark_test`

Reply:
rather strange here...

Reply:
what if `ANewTest extends BaseDataSourceTest` and `override` `$database`?